### PR TITLE
Lint CI

### PR DESCRIPTION
**Pull request by liorsicd**
First lint CI, using flake8,

### CODE REVIEW

File: .github/workflows/python-package.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
```
Comment (ifireball) [inline .github/workflows/python-package.yml:16]: the matrix logic is probably not needed - you're not really planing to support multiple python versions are you?

File: .github/workflows/python-package.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
```
Comment (ifireball) [inline .github/workflows/python-package.yml:26]: you're project is using pipenv to manage dependencies no? you should probably use it here as well

**Review by ifireball — CHANGES_REQUESTED** [review]

File: .github/workflows/python-package.yml (hunk 2 above)
Comment (liorsicd) [reply .github/workflows/python-package.yml:26]: @ifireball the project dependencies are needed here?

File: .github/workflows/python-package.yml (hunk 1 above)
Comment (liorsicd) [reply .github/workflows/python-package.yml]: fixed

File: .github/workflows/python-package.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#        pip install pytest
+    - name: Lint with flake8
+      run: |
+        # stop the build if there are Python syntax errors or undefined names
+        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
+        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
+        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
+#     - name: Test with pytest
+#       run: |
+#         pytest
```
Comment (lioramilbaum) [inline .github/workflows/python-package.yml:34]: Remove those comment lines. Add them when you have tests to be tested by pytest

**Review by lioramilbaum — CHANGES_REQUESTED** [review]

File: .github/workflows/python-package.yml (hunk 2 above)
Comment (liorsicd) [reply .github/workflows/python-package.yml:23]: @ifireball ?

File: .github/workflows/python-package.yml (hunk 3 above)
Comment (liorsicd) [reply .github/workflows/python-package.yml]: @lioramilbaum fixed

**Review by lioramilbaum — APPROVED** [review]

**Pull request merged**